patina_dxe_core/src/event_db.rs

```diff
@@ -608,7 +667,18 @@ impl SpinLockedEventDb {
     ///
     /// Returns r_efi:efi::Status::INVALID_PARAMETER if incorrect parameters are given.
     pub fn signal_event(&self, event: efi::Event) -> Result<(), EfiError> {
-        self.lock().signal_event(event)
+        if let Some(mut guard) = self.try_lock() {
+            guard.signal_event(event)
+        } else {
+            //unable to acquire lock; queue signal for later processing.
+            if let Some(mut pending_signals) = self.pending_signals.try_lock() {
+                pending_signals.push(PendingSignals::Event(event));
+                Ok(())
+            } else {
+                log::warn!("Could not acquire pending signals lock to queue signal.");
```

Comment:
This might be able to be dropped now?

https://github.com/OpenDevicePartnership/patina/blob/9c452663df55b0e11c5cef200f0261274d841311/patina_dxe_core/src/events.rs#L113-L114

This is the one case where we don't acquire a TPL HIGH lock, but also represents an unexpected error condition